library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.ts

```diff
@@ -0,0 +1,279 @@
+
+          this.errorService.handleError(
+            new Error('There was an error creating a bank account')
+          );
+          return of(err);
+        })
+      )
+      .subscribe();
+  }
+
+  createWorkflow(
+    kind: PostWorkflowBankModel.KindEnum
+  ): Observable<WorkflowWithDetailsBankModel> {
+    const poll = new Poll(this.pollConfig);
+    let workflow_guid: string;
+
+    return this.bankAccountService.createWorkflow(kind).pipe(
+      map((workflow) => (workflow_guid = workflow.guid!)),
+      switchMap(() => poll.start()),
+      takeUntil(merge(poll.session$, this.unsubscribe$)),
+      switchMap(() => this.bankAccountService.getWorkflow(workflow_guid)),
+      skipWhile((workflow) => !workflow.plaid_link_token),
+      tap(() => poll.stop())
+    );
+  }
+
+  bootstrapPlaid(linkToken: string): void {
+    combineLatest([
+      this.bankAccountService.getPlaidClient(),
+      this.configService.getConfig$()
+    ])
+      .pipe(
+        take(1),
+        map((obj) => {
+          const [plaidClient, config] = obj;
+          const language = getLanguageFromLocale(config.locale);
+
+          if (!plaidClient) {
+            this.bankAccountService.setPlaidClient(true);
+
+            let script = this._renderer2.createElement('script');
+            script.src = this.plaidScriptSrc;
+            this._renderer2.appendChild(this._document.body, script);
+
+            // Handle error loading the Plaid script
+            script.onerror = () => {
+              this.error$.next(true);
+            };
+
+            script.addEventListener('load', () => {
+              this.plaidCreate(linkToken, language);
+            });
+          } else this.plaidCreate(linkToken, language);
+        }),
+        catchError((err) => {
+          this.error$.next(true);
+          this.eventService.handleEvent(
+            LEVEL.ERROR,
+            CODE.DATA_ERROR,
+            'There was an error launching the Plaid SDK'
+          );
+          this.errorService.handleError(
+            new Error('There was an error launching the Plaid SDK')
+          );
+          return of(err);
+        })
+      )
+      .subscribe();
+  }
+
+  plaidCreate(linkToken: string, language: string) {
+    //@ts-ignore
+    const client = Plaid.create({
+      token: linkToken,
+      language: language,
+      onSuccess: (public_token: string, metadata: any) => {
+        this.plaidOnSuccess(public_token, metadata);
+      },
+      onLoad: () => {
+        client.open();
+      },
+      onExit: (err: any) => this.plaidOnExit(err)
+    });
+  }
+
+  plaidOnSuccess(public_token: string, metadata?: any): void {
+    function isValidAsset(asset: string | null | undefined) {
+      return typeof asset == 'string';
+    }
+    function isOnlyAccount(accounts: any[]) {
+      return accounts.length == 1;
+    }
+
+    // const asset = 'USD';
+    const asset = metadata.accounts[0].iso_currency_code;
+    const account = metadata.accounts[0];
```

Comment:
You'll want to guard this with the `isOnlyAccount` call so you don't get an error referencing an element in an empty list (or null list)

Reply:
Good idea. I've moved them into the if statement.